### PR TITLE
[CI/Build] Add health check

### LINKED ISSUE #4867: [Feature]: Health check for restart policy
### 🚀 The feature, motivation and pitch

A small change to get alot of reliability back....

We see vLLM crash or hang in various ways, e.g.:

https://github.com/vllm-project/vllm/issues/4108
https://github.com/vllm-project/vllm/issues/4344

And manually managing that is a hassle.

vLLM team could easily add a HEALTHCHECK line in the Dockerfile so tools like autoheal can function.

https://hub.docker.com/r/willfarrell/autoheal/
https://docs.docker.com/reference/dockerfile/#healthcheck

Would looks like:
```
HEALTHCHECK --interval=5m --timeout=10s curl -f http://localhost/health || exit 1
```

This would allow one to use the other docker image to manage the vLLM images.

### Alternatives

Manual labor

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pseudotensor**
https://hub.docker.com/r/willfarrell/autoheal/
https://docs.docker.com/reference/dockerfile/#healthcheck

To deal with issues like this:

https://github.com/vllm-project/vllm/issues/4108
https://github.com/vllm-project/vllm/issues/4344

FIX #4867

### CODE REVIEW

**Comment by robertgshaw2-neuralmagic** [comment]
Can you explain a bit about what `HEALTHCHECK` does?

**Comment by pseudotensor** [comment]
> Can you explain a bit about what `HEALTHCHECK` does?

It's explained in those links.  But basically, if one uses autoheal docker image to monitor other images that have the HEALTHCHECK present, then that autoheal will restart the image if the health condition fails.  Often vLLM fails, but the entire image still goes on in unhealthy state.

File: Dockerfile [hunk 1]
```diff
@@ -138,5 +138,10 @@ RUN --mount=type=cache,target=/root/.cache/pip \
 
 ENV VLLM_USAGE_SOURCE production-docker-image
 
+# https://docs.docker.com/reference/dockerfile/#healthcheck
+# https://hub.docker.com/r/willfarrell/autoheal/
+HEALTHCHECK --interval=5m --timeout=10s \
+ CMD curl -f http://localhost/health || exit 1
```
Comment (mgoin) [inline Dockerfile:144]: nit but 4 spaces indent is standard for this file
```suggestion
    CMD curl -f http://localhost/health || exit 1
```

File: Dockerfile [hunk 2]
```diff
@@ -138,5 +138,10 @@ RUN --mount=type=cache,target=/root/.cache/pip \
 
 ENV VLLM_USAGE_SOURCE production-docker-image
 
+# https://docs.docker.com/reference/dockerfile/#healthcheck
+# https://hub.docker.com/r/willfarrell/autoheal/
```
Comment (mgoin) [inline Dockerfile:142]: Please leave a written out comment before sharing these links

**Pull request closed without merging**